pkg/httphelper/httphelper.go

```diff
@@ -6,14 +6,30 @@ import (
 	"net/http"
 )
 
-func Error(w http.ResponseWriter, err error) {
+type JSONError struct {
+	Code    int         `json:"code"`
+	Message string      `json:"message"`
+	Detail  interface{} `json:"detail,omitempty"`
```

Comment:
How do we expect to use this `Detail` field?  Is it just a stringifiable, and the reciever is expected to be able to display it but probably not parse it?

I'm not sure if this will apply to the intended use pattern, but are you familiar with http://golang.org/pkg/encoding/json/#RawMessage ?  It can be useful when you have a "generic" field with a type hint field next to it; you can defer deserializing it on the first pass, then do a switch on the type hint to another deserialize call that expects a concrete type.